src/Forcings/model_forcing.jl

```diff
@@ -0,0 +1,49 @@
+using Oceananigans.Utils: with_tracers
+using Oceananigans.Fields: assumed_field_location
+
+@inline zeroforcing(args...) = 0
+
+# `regularize_forcing` is used to "regularize", or add information to,
+# user-defined forcing objects that are passed to the `IncompressibleModel`
+# constructor. In particular, it is only until `model_forcing` is called that the
+# *names* of various forcing fields are available. The field_name can be used to infer
+# the location at which the forcing is applied, or to add a field dependency
+# to a special forcing object, as for Relxation.
+
+regularize_forcing(forcing, field_name) = forcing # fallback
```

Comment:
Why not make this a docstring?

Reply:
Hmm, this comment is about the code structure, and not how the code works... but I suppose it could still be a docstring. I just thought that this point is irrelevant to users. What do you think?

We have made some decisions in `IncompressibleModel` that are convenient for users but make the internal code a bit harder to understand, so comments / docstrings like these are important, I think.